.github/workflows/publish.yaml

```diff
@@ -0,0 +1,126 @@
+        run: |
+          echo "matrix=$(docker buildx bake ${{ env.BAKE_GROUP }} -f ${{ env.BAKE_FILE }} --print | jq -cr '.group.default.targets')" >> $GITHUB_OUTPUT
+
+      # 1.3 (optional) - output the generated target list for verification
+      - name: Show matrix
+        run: |
+          echo ${{ steps.targets.outputs.matrix }}
+
+  build-push:
+    name: build-bake-push
+    runs-on: ubuntu-22.04
+    if: github.ref_name == 'main' || contains(github.event.ref, '/tags/')
+    permissions:
+      packages: write
+      contents: read
+
+    # this job depends on the following jobs before beginning
+    needs:
+      - targets
+      - correctness
+      - formatting
+
+    # 2.0 - Build a matrix strategy from the retrieved target list
+    strategy:
+      fail-fast: true
+      matrix:
+        target: ${{ fromJson(needs.targets.outputs.matrix) }}
+
+    steps:
+      # 2.1 - Checkout the repository
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.workflow_run.head_branch }}
+
+      # 2.2 - Set environment variables to be used within the bake file
+      # NOTE: this overrides the defaults and should be specific to this deployment
+      - name: set env variables for bake file
+        run: |
+          if [[ '${{ github.ref_type }}' == 'branch' && '${{ github.ref_name }}' == 'main' ]]; then
+            TAG=latest
+          else
+            SEMVER=$( echo ${{ github.ref_name }} | sed -nre 's/^v[^0-9]*(([0-9]+\.)*[0-9]+(-[a-z]+)?).*/\1/p')
+            if [[ -n $SEMVER ]]; then
+              TAG=${SEMVER}
+            else
+              TAG=${{ github.ref_name }}
+            fi
+          fi
+
+          echo "$TAG"
+
+          echo "VERSION=${TAG,,}" >> ${GITHUB_ENV}
+          echo "DOCKER_REGISTRY=${{ env.REGISTRY }}" >> ${GITHUB_ENV}
+          # Lowercase the owner for docker image naming conventions
+          echo "DOCKER_ORG=${OWNER,,}" >> ${GITHUB_ENV}
+          java --version
+
+      # 2.3 - Login against the docker registry
+      - name: Login to registry ${{ env.REGISTRY }}
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # 2.4 - Setup QEMU for platform emulation
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      # 2.5 - Setup Docker BuildX for multi platform building
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      # 2.6 - Build and push Docker Images
+      - name: Build Images using BuildX Bake
+        uses: docker/bake-action@v2
+        with:
+          files: ./${{ env.BAKE_FILE }}
+          targets: ${{ matrix.target }}
+          push: true
+          set: |
+            *.cache-from=type=gha,scope=build-${{ matrix.target }}
+            *.cache-to=type=gha,scope=build-${{ matrix.target }},mode=max
```

Comment:
Suggest using the following to abstract the bake process further.
There are some limitations but should be sufficient for this case.

```suggestion
  # Generate a more robust tag for the image
  tag-generator:
    name: Determine image tag
    runs-on: ubuntu-22.04
    outputs:
      image-tag: ${{ steps.generate.outputs.tag }}
    steps:
      - name: Generate apprropriate tag
        id: generate
        run: |
          if [[ '${{ github.ref_type }}' == 'branch' && '${{ github.ref_name }}' == 'main' ]]; then
            TAG=latest
          else
            SEMVER=$( echo ${{ github.ref_name }} | sed -nre 's/^v[^0-9]*(([0-9]+\.)*[0-9]+(-[a-z]+)?).*/\1/p')
            if [[ -n $SEMVER ]]; then
              TAG=${SEMVER}
            else
              TAG=${{ github.ref_name }}
            fi
          fi

          echo "$TAG"
          echo "tag=${TAG,,}" >> ${GITHUB_OUTPUT}

      - name: Show Generated Tag
        run: echo ${{ steps.generate.outputs.tag }}

  # Build and Publish all targets associated with specified group
  bake:
    needs:
      - correctness
      - formatting
      - tag-generator
    uses: darpa-askem/orchestration/.github/workflows/bake-publish.yml@d34cc635c0729edc12fc6bf0f2bffd07e585316b
    with:
      file: 'docker-bake.hcl'
      group: 'prod'
      registry: 'ghcr.io'
      organization: ${{ github.repository_owner }}
      tag: ${{ needs.tag-generator.outputs.image-tag }}
    secrets:
      username: ${{ github.repository_owner }}
      password: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
fixed